torch/autograd/gradcheck.py

```diff
@@ -302,16 +320,45 @@ def fn(input):
         analytical, reentrant, correct_grad_sizes = get_analytical_jacobian(tupled_inputs, o, nondet_tol=nondet_tol)
         numerical = get_numerical_jacobian(fn, tupled_inputs, eps=eps)
 
-        if not correct_grad_sizes:
+        out_is_complex = o.is_complex()
+
+        if out_is_complex:
+            # analytical vjp with v = 1.0j
+            analytical_with_imag_v, reentrant_with_imag_v, correct_grad_sizes_ \
+                = get_analytical_jacobian(tupled_inputs, o, nondet_tol=nondet_tol, v=1j)
+
+        if not correct_grad_sizes or (out_is_complex and not correct_grad_sizes_):
             return fail_test('Analytical gradient has incorrect size')
 
-        for j, (a, n) in enumerate(zip(analytical, numerical)):
-            if a.numel() != 0 or n.numel() != 0:
-                if not torch.allclose(a, n, rtol, atol):
-                    return fail_test('Jacobian mismatch for output %d with respect to input %d,\n'
-                                     'numerical:%s\nanalytical:%s\n' % (i, j, n, a))
+        def checkIfNumericalAnalyticAreClose(a, n, error_str=''):
+            if not torch.allclose(a, n, rtol, atol):
+                return fail_test(error_str + 'Jacobian mismatch for output %d with respect to input %d,\n'
+                                 'numerical:%s\nanalytical:%s\n' % (i, j, n, a))
 
-        if not reentrant:
+        inp_tensors = iter_tensors(input, True)
+
+        for j, (a, n, inp) in enumerate(zip(analytical, numerical, inp_tensors)):
+            if a.numel() != 0 or n.numel() != 0:
+                if inp.is_complex():
+                    # C -> R, C -> C
+                    checkIfNumericalAnalyticAreClose(a, n[:, :, 0] + n[:, :, 1].conj(),
+                                                     "Real value(s) of Conjugate Wirtinger derivative \
+                                                     (Re(ds/dx + ds/dy * 1j)) failed to compare equal.")
```

Comment:
nit: missing a space at the end to concatenate properly with the other string no?